File: drs-component/drs-message/src/main/scala/com/kindminds/drs/api/message/command/Logistics.scala
package com.kindminds.drs.api.message.command


package ivs {

  import com.kindminds.drs.api.message.Command
  import com.kindminds.drs.api.v2.biz.domain.model.logistics.ShipmentStatus


  case class InitVerifyIvsProductInfo(ivsNumber:String , boxNum:Int, mixedBoxLineSeq: Int) extends Command

  case class ConfirmIvsProductVerifyInfo(ivsNumber:String , boxNum:Int, mixedBoxLineSeq: Int , docReq : String) extends Command

  case class RejectIvsProductVerifyInfo(ivsNumber:String , boxNum:Int, mixedBoxLineSeq: Int , docReq : String) extends Command

  case class UpdateIvsStatus(ivsNumber:String , status : Option[ShipmentStatus],
                             sku:String, boxNum:Int) extends  Command


}

File: drs-component/drs-common/src/main/java/com/kindminds/drs/api/v2/biz/domain/model/Logistics.java
package com.kindminds.drs.api.v2.biz.domain.model;

import java.util.Date;

import com.kindminds.drs.Warehouse;
import com.kindminds.drs.api.v1.model.replenishment.ReplenishmentTimeSpent;

public interface Logistics {

	public Date calculateRecommendedFcaDeliveryDate(Date expectedStockDepletionDate, ReplenishmentTimeSpent timeSpent, Warehouse warehouse);
	public Date calculateEstimatedReceivingCompletionDate(Date expectedStockDepletionDate, ReplenishmentTimeSpent timeSpent,Warehouse warehouse);
	
}
